src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -178,6 +188,162 @@ protected boolean doCreateTableIfNotExists(
+                        collect(Collectors.joining(",")) + "}";
+                String entityOthers = "{" + entityOthersMap.entrySet().stream()
+                        .map(e -> e.getKey() + " " + e.getValue()).
+                        collect(Collectors.joining(",")) + "}";
+
+                String msg;
+                // check shard keys and types match
+                if (!tableShards.equals(entityShards)) {
+                    msg = String.format("Shard primary keys mismatch: " +
+                            "table=%s, entity=%s.", tableShards, entityShards);
+                    errors.add(msg);
+                }
+
+                // check non-shard keys and types match
+                if (!tableNonShards.equals(entityNonShards)) {
+                    msg = String.format("Non-shard primary keys mismatch: " +
+                                    "table=%s, entity=%s.", tableNonShards,
+                            entityNonShards);
+                    errors.add(msg);
+                }
+
+                // check non-primary keys and types match
+                if (!tableOthers.equals(entityOthers)) {
+                    msg = String.format("Non-primary key columns mismatch:" +
+                                    "table=%s, entity=%s.", tableOthers,
+                            entityOthers);
+                    errors.add(msg);
+                }
+
+                // check identity same
+                FieldValue identity = tableSchema.get(identityField);
+                if (identity != null && !entityInformation.isAutoGeneratedId()) {
+                    errors.add("Identity information mismatch.");
+
+                } else if (identity == null && entityInformation.isAutoGeneratedId() &&
+                        entityInformation.getIdNosqlType() != FieldValue.Type.STRING) {
+                    errors.add("Identity information mismatch.");
+                }
+
+                // TTL warning
+                FieldValue ttlValue = tableSchema.get(ttlField);
+                TimeToLive ttl = entityInformation.getTtl();
+                // TTL is present in database but not in the entity
+                if (ttlValue != null && ttl != null &&
+                        !ttl.toString().equalsIgnoreCase(ttlValue.getString())) {
+                    LOG.warn("TTL of the table in database is different from " +
+                            "the TTL of the entity " +
+                            entityInformation.getJavaType().getName());
+                } else if (ttlValue == null && ttl != null && ttl.getValue() != 0) {
+                    // TTL is present in entity but not in the database
+                    LOG.warn("TTL of the table in database is different from " +
+                            "the TTL of the entity " +
+                            entityInformation.getJavaType().getName());
+                }
+            }
+        } catch (Exception e) {
```

Comment:
I catched the generic exception here to avoid any errors during parsing of the JSON table schema. If at all JSON schema is changed it may result in nullpointer exception and user can't use the driver. IMO this should not stop user from using the driver.  exception can be changed to nullpointerexception.

Reply:
Modified to catch only NullPointerException